### DISCUSSION TITLE
Explain file types in reosetta source

### DISCUSSION THREAD

**Discussion by Teyiowuawi** [finos/common-domain-model#3268] [Q&A] [answered]
Hello! 

Is it possible to give a brief description of the different filetypes within the Rosetta source folder of the CDM project?

The .type and .enum files makes sense to interpret as classes, their properties and enumerations. 

How about the logic in the .func, .synonym files? 

Wuuld be appreciated if someone understood this!

**Comment by lolabeis** [comment, marked as answer]
Hi @Teyiowuawi

Same as the `.type` and `.enum` files containing types and enums, `.func` files are designed to contain functions (prefixed with `func` in the DSL), whereas `.synonym` files contain the model-to-model mappings (known as `synonym` in the DSL).

You can find a fuller explanation of the different modelling components available in the DSL that supports the CDM (which has been renamed as "Rune" and is also hosted at FINOS) here:
https://docs.rosetta-technology.io/rosetta/rosetta-dsl/rune-modelling-component/

**Comment by CDM-ReleaseManagement-LSG** [comment]
I hope the previous answer helped but I will close this for now.  Please open another Discussion or Issue, if you wish.
